src/Development/IDE/Core/Rules.hs

```diff
@@ -392,7 +393,8 @@ rawDependencyInformation fs = do
       -- If we have, just return its Id but don't update any of the state.
       -- Otherwise, we need to process its imports.
       checkAlreadyProcessed f $ do
-          al <- lift $ modSummaryToArtifactsLocation f <$> use_ GetModSummaryWithoutTimestamps f
+          msum <- lift $ use GetModSummaryWithoutTimestamps f
```

Comment:
We don't want to fail here, since the result of `GetModIface` now depends on the entire graph's headers/modsummaries

Reply:
If even one module header failed to parse here, `NeedsObjCode` would fail, which would make typechecking fail for everything in the entire project.

Reply:
I see, because of the `GetModuleGraph` dependency. So you have made this code able to tolerate failures. Fair enough